File: TeddyPet_BE/src/main/java/fpt/teddypet/application/dto/request/user/UserAddressRequest.java
package fpt.teddypet.application.dto.request.user;

import jakarta.validation.constraints.NotBlank;
import lombok.Builder;

@Builder
public record UserAddressRequest(
        @NotBlank(message = "Họ tên là bắt buộc") String fullName,

        @NotBlank(message = "Số điện thoại là bắt buộc") String phone,

        @NotBlank(message = "Địa chỉ là bắt buộc") String address,

        Double longitude,
        Double latitude,
        boolean isDefault) {
}

File: teddypet_mobile/lib/data/models/request/user/user_address_request.dart
class UserAddressRequest {
  final String fullName;
  final String phone;
  final String address;
  final double? longitude;
  final double? latitude;
  final bool isDefault;

  UserAddressRequest({
    required this.fullName,
    required this.phone,
    required this.address,
    this.longitude,
    this.latitude,
    this.isDefault = false,
  });

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'phone': phone,
      'address': address,
      'longitude': longitude,
      'latitude': latitude,
      'isDefault': isDefault,
    };
  }
}
